### PR TITLE
accept config parameters while testing performance

### PR DESCRIPTION
**Pull request by kakazuo**
let method _performanceBudgetTest_ accept parameter while being invoked,  remove unused package rimraf and file main.js

### CODE REVIEW

File: lib/sitespeed-utils.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ var common = require('./common')
 var Mocha = require('mocha');
 var fs = require('fs');
 var path = require('path');
+spUtilsCfg = {};
```
Comment (aerwin) [inline lib/sitespeed-utils.js:6]: @kakazuo --- Are you missing a `var` here?

File: test/test.js [hunk 2]
```diff
@@ -72,14 +39,17 @@ function buildUrl(baseUrl, urlParams, pathname) {
 
 describe('Sitespeed Budgets', function() {
     // Need to bump up time for the Sitespeed tests
-    this.timeout(60000);
+    this.timeout(60000 * 2);
```
Comment (aerwin) [inline test/test.js]: @kakazuo I suppose I'm the one that hard-coded 60000 to start with, but could we possibly make this a constant? And  maybe even configurable?

File: test/test.js [hunk 3]
```diff
@@ -4,44 +4,11 @@ var url = require('url');
...
-/*rimraf('sitespeed-result', function(err) {
-    console.log("delete legacy test data success")
-});
-*/
-
-var path1 = __dirname + '/../../../config/budget-testcases.json';
-var path2 = __dirname + '/../config/budget-testcases.json';
-var config = process.env.BUDGET_TESTCASES;
-if (!config) {
-    if (fs.existsSync(path1)) {
```
Comment (aerwin) [inline test/test.js:20]: @kakazuo I'm surprised you removed all of the stuff with loading the config from a file? Could you explain? I was never opposed to that... the comment in the other pull request was that I didn't want Bluemix-specific logic in this module for retrieving Bluemix login cookies.

File: test/test.js [hunk 4]
```diff
@@ -4,44 +4,11 @@ var url = require('url');
...
-            process.exit(1);
-        }
-    } else {
-        console.error('please configure test case file in ' + path1 + ' or ' + path2);
-        process.exit(1);
-    }
-}
-var baseUrl = config.baseUrl;
-var testCases = config.testCases;
+var baseUrl = spUtilsCfg.baseUrl;
```
Comment (aerwin) [inline test/test.js:10]: @kakazuo So, per my https://github.com/aerwin/sitespeed-utils/pull/4/commits/a75afe394fce80d16d17987d09f2d3f9ae89d0a5#r61295134, I guess you must be relying on `spUtilsCfg` being a global variable? Isn't there some way to pass it in when you create the `Mocha` object?

File: test/test.js (hunk 2 above)
Comment (kakazuo) [inline test/test.js]: sure, I will update this part

File: test/test.js (hunk 3 above)
Comment (kakazuo) [inline test/test.js:20]: Yeah, I understood your meaning about move bleumix related logic out of this package, here I just want let each caller load config file by themselves, so they can put all parameters in one config file(like username and password which is needed by cookie getting),  and caller still can add other parameter to this config object dynamically,  eg: add a request header after getting the cookie,  so it should be more flexible. Meanwhile, the whole process just need to load config file from one place, it makes logic of config file loading more simple as well.  what do you think?

File: test/test.js (hunk 4 above)
Comment (kakazuo) [inline test/test.js:10]: I need to investigate this, try to find a way to pass the parameter, not use global variable

File: test/test.js (hunk 3 above)
Comment (aerwin) [inline test/test.js:20]: @kakazuo Ok, sure... I can see that argument... maybe we should at least provide a little code snippet in the README that shows an example of loading that file and calling `performBudgetTest`

Comment (kakazuo) [inline test/test.js:20]: ok, good point

File: test/test.js (hunk 4 above)
Comment (kakazuo) [inline test/test.js:10]: @aerwin I read the source code of mocha project,  did not find appropriate method to pass config object,  so I am thinking if we can merge this commit first and investigate it later continuously

**Pull request merged**